lib/msf/base/simple/statistics.rb

```diff
@@ -16,73 +16,53 @@ class Statistics
   #
   def initialize(framework)
     self.framework = framework
+    Msf::Modules::Metadata::Cache.instance.update_stats
   end
 
   #
   # Returns the number of encoders in the framework.
   #
   def num_encoders
-    framework.encoders.length
+    Msf::Modules::Metadata::Cache.instance.module_counts[:encoder]
   end
 
   #
   # Returns the number of exploits in the framework.
   #
   def num_exploits
-    framework.exploits.length
+    Msf::Modules::Metadata::Cache.instance.module_counts[:exploit]
   end
 
   #
   # Returns the number of NOP generators in the framework.
   #
   def num_nops
-    framework.nops.length
+    Msf::Modules::Metadata::Cache.instance.module_counts[:nop]
   end
 
   #
   # Returns the number of payloads in the framework.
   #
   def num_payloads
-    framework.payloads.length
+    Msf::Modules::Metadata::Cache.instance.module_counts[:payload]
   end
 
   #
   # Returns the number of auxiliary modules in the framework.
   #
   def num_auxiliary
-    framework.auxiliary.length
+    Msf::Modules::Metadata::Cache.instance.module_counts[:auxiliary]
   end
 
   #
   # Returns the number of post modules in the framework.
   #
   def num_post
-    framework.post.length
+    Msf::Modules::Metadata::Cache.instance.module_counts[:post]
   end
 
   def num_evasion
-    framework.evasion.length
-  end
-
-  #
-  # Returns the number of stages in the framework.
-  #
-  def num_payload_stages
-    framework.payloads.stages.length
-  end
-
-  #
-  # Returns the number of stagers in the framework.
-  #
-  def num_payload_stagers
-    framework.payloads.stagers.length
-  end
-
-  #
-  # Returns the number of singles in the framework.
-  #
```

Comment:
I've removed these as they weren't being referenced anywhere and when I checked the values they had on master they were incorrect anyway